File: ios/Objective-C/TinyBarcode/TinyBarcode/Views/CameraZoomFloatingButton.m
#import "CameraZoomFloatingButton.h"

@interface CameraZoomFloatingButton ()

@property (nonatomic, strong) UILabel *zoomLabel;

@end

@implementation CameraZoomFloatingButton


- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        [self setupUI];
    }
    return self;
}

- (void)setupUI {
    self.backgroundColor = [UIColor clearColor];
    self.layer.cornerRadius = self.height / 2.0;
    
    self.zoomLabel = [[UILabel alloc] initWithFrame:CGRectMake(0, 0, self.width, self.height)];
    self.zoomLabel.backgroundColor = [[UIColor blackColor] colorWithAlphaComponent:0.42];
    self.zoomLabel.textColor = [UIColor whiteColor];
    self.zoomLabel.layer.cornerRadius = self.height / 2.0;
    self.zoomLabel.layer.masksToBounds = YES;
    self.zoomLabel.font = kFont_SystemDefault(kCameraZoomFloatingLabelTextSize);
    self.zoomLabel.textAlignment = NSTextAlignmentCenter;
    [self addSubview:self.zoomLabel];
    
    UITapGestureRecognizer *tapGes = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(tapGesAction)];
    [self addGestureRecognizer:tapGes];
    
}

// MARK: - Setter
- (void)setCurrentCameraZoom:(CGFloat)currentCameraZoom {
    _currentCameraZoom = currentCameraZoom;
    self.zoomLabel.text = [NSString stringWithFormat:@"%.1fx", currentCameraZoom];
}

- (void)tapGesAction {
    if (self.tapPressActionBlock) {
        self.tapPressActionBlock();
    }
}


@end

File: ios/Swift/TinyBarcode/TinyBarcode/Views/CameraZoomFloatingButton.swift
//
//  CameraZoomFloatingButton.swift
//  TinyBarcode
//
//  Created by dynamsoft's mac on 2022/11/24.
//

import UIKit

typealias TapPressCompletion = () -> Void

class CameraZoomFloatingButton: UIView {
    
    private var cameraZoom: CGFloat = kDCEDefaultZoom
    
    var tapPressCompletion: TapPressCompletion?
    
    var currentCameraZoom: CGFloat {
        get {cameraZoom}
        set {
            cameraZoom = newValue
            self.zoomLabel.text = String(format: "%.1fx", cameraZoom)
        }
    }
    
    private lazy var zoomLabel: UILabel = {
        let label = UILabel.init(frame: CGRectMake(0, 0, self.width, self.height))
        label.backgroundColor = .black.withAlphaComponent(0.42)
        label.textColor = .white
        label.layer.cornerRadius = self.height / 2.0;
        label.layer.masksToBounds = true
        label.font = kFont_SystemDefault(kCameraZoomFloatingLabelTextSize)
        label.textAlignment = .center
        return label
    }()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        setupUI()
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
    }
    
    private func setupUI() -> Void {
        self.addSubview(zoomLabel)
        let tapGes = UITapGestureRecognizer.init(target: self, action: #selector(tapGesAction(_:)))
        self.addGestureRecognizer(tapGes)
    }
    
    @objc private func tapGesAction(_ tapGes: UITapGestureRecognizer) -> Void {
        self.tapPressCompletion?()
    }

}
